eclass/nodejs.eclass

```diff
@@ -0,0 +1,305 @@
+    # shellcheck disable=SC2185
+    find -type f -name docker-compose.yml -delete || die
+    # shellcheck disable=SC2185
+    find -type f -iname CopyrightNotice.txt -delete || die
+
+    # Remove misc files
+    # shellcheck disable=SC2185
+    find -type f -iname '*.musl.node' -delete || die
+    # shellcheck disable=SC2185
+    find -type f -iregex '.*\.\(editorconfig\|bak\|npmignore\|exe\|gitattributes\|ps1\|ds_store\|log\|pyc\)$' -delete || die
+    # shellcheck disable=SC2185
+    find -type f -iregex '.*\.\(travis.yml\|makefile\|jshintrc\|flake8\|mk\|env\|nycrc\|eslint.*\|coveralls.*\)$' -delete || die
+    # shellcheck disable=SC2185
+    find -type f -iregex '.*\.\(jscs.json\|jshintignore\|gitignore\|babelrc.*\|runkit_example.js\|airtap.yml\)$' -delete || die
+    # shellcheck disable=SC2185
+    find -type f -iregex '.*\.\(jekyll-metadata\|codeclimate.yml\|prettierrc.yaml\|drone.jsonnet\|mocharc.*\)$' -delete || die
+    # shellcheck disable=SC2185
+    find -type f -iname makefile -delete || die
+    # shellcheck disable=SC2185
+    find -type f -name '*\~' -delete || die
+
+    # shellcheck disable=SC2185
+    find -type d \
+        \( \
+        -iwholename '*.github' -o \
+        -iwholename '*.tscache' -o \
+        -iwholename '*.vscode' -o \
+        -iwholename '*.idea' -o \
+        -iwholename '*.nyc_output' -o \
+        -iwholename '*.deps' -o \
+        -iwholename '*/man' -o \
+        -iwholename '*/test' -o \
+        -iwholename '*/scripts' -o \
+        -iwholename '*/git-hooks' -o \
+        -iwholename '*/prebuilds' -o \
+        -iwholename '*/android-arm' -o \
+        -iwholename '*/android-arm64' -o \
+        -iwholename '*/linux-arm64' -o \
+        -iwholename '*/linux-armvy' -o \
+        -iwholename '*/linux-armv7' -o \
+        -iwholename '*/linux-arm' -o \
+        -iwholename '*/win32-ia32' -o \
+        -iwholename '*/win32-x64' -o \
+        -iwholename '*/darwin-x64' \
+        -iwholename '*/darwin-x64+arm64' \
+        \) \
+        -exec rm -rvf {} +
+}
+
+# @FUNCTION: enpm
+# @DESCRIPTION:
+# Packet manager execution wrapper
+enpm() {
+    debug-print-function "${FUNCNAME}" "${@}"
+
+    local mynpmflags_local mynpmflagstype npmflags
+
+    # Make the array a local variable since <=portage-2.1.6.x does not support
+    # global arrays (see bug #297255). But first make sure it is initialised.
+    [[ -z ${mynpmflags} ]] && declare -a mynpmflags=()
+    mynpmflagstype=$(declare -p mynpmflags 2>&-)
+    if [[ "${mynpmflagstype}" != "declare -a mynpmflags="* ]]; then
+        die "mynpmflags must be declared as array"
+    fi
+
+    mynpmflags_local=("${mynpmflags[@]}")
+
+    npmflags=(
+        --color false
+        --foreground-scripts
+        --offline
+        --progress false
+        --verbose
+        "${mynpmflags_local[@]}"
+    )
+
+    case ${NODEJS_MANAGEMENT} in
+    npm)
+        npmflags+=(
+            --audit false
+        )
+        npm "$@" "${npmflags[@]}"
```

Comment:
Missing dies again.

Reply:
I had not included the day within the enpm function, so it was included outside the function and thus included the appropriate error message

look when it is used in pm, example:

`https://github.com/gentoo/gentoo/pull/33426/eclass/nodejs.eclass#L256-L260`